src/compiler/zx_calculus.jl

```diff
@@ -1,14 +1,72 @@
-using ZXCalculus
+using ZXCalculus, LightGraphs
+using ZXCalculus: qubit_loc
 
 function optimize!(ir::YaoIR)
     circ = to_ZX_diagram(ir)
-    clifford_simplify!(circ)
-    ir = to_YaoIR(circ)
+    circ = clifford_simplify(circ)
+    ir.body = to_YaoIR(circ)
     return ir
 end
 
+function to_YaoIR(circ::ZXDiagram{T, P}) where {T, P}
+    lo = circ.layout
+    vs = spiders(circ)
+    locs = Dict()
+    nqubit = lo.nbits
+    frontier_v = ones(T, nqubit)
+    ex = quote
+        $(Expr(:meta, :register, :new, gensym(:register)))
+        return
+    end
+    lowered_ast = Meta.lower(@__MODULE__, ex)
```

Comment:
you don't have to go through the parser again. you can use `IR()` to construct an empty IR object directly, and then insert SSA statements.